tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3736,18 +3920,32 @@ private[optimizer] object OptimizerCore {
       }
     }
 
-    def unapply(preTrans: PreTransBlock): Some[(List[BindingOrStat], PreTransLocalDef)] =
+    def unapply(preTrans: PreTransBlock): Some[(List[BindingOrStat], PreTransResult)] =
       Some(preTrans.bindingsAndStats, preTrans.result)
   }
 
-  private sealed abstract class PreTransNoBlock extends PreTransform
+  private sealed abstract class PreTransResult extends PreTransform
```

Comment:
Could you start documenting the different subtypes of pretransforms? Especially the difference between `PreTransResult` and `PreTransGenTree` seems (from a mere naming perspective) unclear to me.